task_16/test/specs/main.js

```diff
@@ -0,0 +1,221 @@
+const Helper = require('../helper/Helper')
+const Main = require('../pageobjects/main.page')
+const Email = require('../pageobjects/email.page')
+
+describe('Hard core', () => {
+    let tempEmail, 
+        searchBtn,
+        googleWindow,
+        emailWindow
+
+    it('Open Google Cloud Homepage', async () => {
+       
+        //Open the url
+        await browser.url('/')
+        await expect(browser).toHaveTitleContaining('Google Cloud')
+    })
+    
+    it('Find and click on search button', async ()=> {
+        //Select search button
+        searchBtn = await Main.searchBtn
+        await Helper.waitExist(searchBtn)
+        await searchBtn.click()
+
+        await expect(searchBtn).toExist()
+    })
+
+    it('Enter search query', async ()=>{
+        //Insert search query into the search bar
+        await searchBtn.setValue(Main.query)
+        await browser.keys("\uE007")
+        
+        await expect(searchBtn).toHaveValue(Main.query)
+    })
+
+    it("Click on the necessary result", async ()=> {
+        //Wait until DOM loads new content
+        browser.waitUntil(() => browser.execute(() => document.readyState === 'complete'),
+            {
+                timeout: 60 * 1000, timeoutMsg: 'Failed to load DOM within a timeframe'
+            }
+        )
```

Comment:
Please try to use this approach: https://webdriver.io/docs/timeouts/#session-page-load-timeout